app/src/main/java/com/burhanrashid52/photoediting/FileSaveHelper.java

```diff
@@ -0,0 +1,130 @@
+package com.burhanrashid52.photoediting;
+
+import android.content.ContentResolver;
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.MediaStore;
+
+import androidx.annotation.RequiresApi;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.lifecycle.Lifecycle;
+import androidx.lifecycle.LifecycleObserver;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.OnLifecycleEvent;
+
+import java.io.OutputStream;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class FileSaveHelper implements LifecycleObserver {
+    private ContentResolver contentResolver;
+    private ExecutorService executor;
+    private MutableLiveData<FileCreateData> fileCreatedResult;
+    private OnFileCreateResult resultListener;
+    private final Observer<FileCreateData> observer = fileCreateData -> {
+        if (resultListener != null) {
+            resultListener.onFileCreateResult(fileCreateData.isCreated,
+                    fileCreateData.filePath,
+                    fileCreateData.error,
+                    fileCreateData.uri);
+        }
+    };
+
+    public FileSaveHelper(AppCompatActivity context) {
+        if (isSdk29OrHigher()) {
+            executor = Executors.newSingleThreadExecutor();
+            this.contentResolver = context.getContentResolver();
+            fileCreatedResult = new MutableLiveData<>();
+            fileCreatedResult.observe(context, observer);
+            context.getLifecycle().addObserver(this);
```

Comment:
Since we are using context just `addObserver` to lifecycle, then I would recommend this to move this logic to the `EditImageActivity` and remove the context from the constructor.

Reply:
will change that .